### DISCUSSION TITLE
Need help with loadImage Nodejs

### DISCUSSION THREAD

**Discussion by Frozzesav** [fabricjs/fabric.js#9899] [Q&A] [answered]
Hello! I'm try to load image, but I can't undestand how to do this.
`
import http from 'http';
import * as fabric from 'fabric/node'; // v6

const port = 8080;

http
  .createServer((req, res) => {
    const canvas = new fabric.StaticCanvas(null, { width: 595, height: 842, });
    const rect = new fabric.Rect({ width: 595, height: 500, fill: '#ff0000' });

    fabric.util.loadImage("http://fabricjs.com/assets/pug.jpg", function (img) {
      var oImg = new fabric.FabricImage(img);
      console.log(img.src)
      oImg.set({
        left: 0,
        top: 0,
        width: 1000,
        height: 1000,
        originX: 'center',
        originY: 'center',
        transparentCorners: false
      });
      canvas.add(oImg);
      // canvas.renderAll();
    })
    canvas.renderAll();

    const handleResponse = (dataURL) => {
      if (req.url === '/download') {
        res.setHeader('Content-Type', 'image/png');
        res.setHeader('Content-Disposition', 'attachment; filename="fabric.png"');
        const base64Data = dataURL.replace(/^data:image\/png;base64,/, "");
        const buffer = Buffer.from(base64Data, 'base64');
        res.end(buffer);
      } else if (req.url === '/view') {
        res.setHeader('Content-Type', 'image/png');
        const base64Data = dataURL.replace(/^data:image\/png;base64,/, "");
        const buffer = Buffer.from(base64Data, 'base64');
        res.end(buffer);
      } else {
        res.writeHead(200, { 'Content-Type': 'text/html' });
        res.write(`<img src="${dataURL}" />`);
        res.end();
      }
    };

    const dataURL = canvas.toDataURL({ multiplier: 4 });
    handleResponse(dataURL);
  })
  .listen(port, (err) => {
    if (err) throw err;
    console.log(
      `> Ready on http://localhost:${port}, http://localhost:${port}/view, http://localhost:${port}/download`
    );
  });
`

**Comment by asturur** [comment, marked as answer]
fabric.util.loadImage returns a promise, does not take a callback anymore
